README.md

```diff
@@ -1,27 +1,51 @@
 # SWI-Prolog: A comprehensive Prolog implementation
 
-## Forking, cloning and submitting patches
+## Getting the source code
 
 This repositories uses many [GIT
-submodules](https://git-scm.com/book/en/v1/Git-Tools-Submodules). This
-cause the common __fork and clone not to work__. Instead, _clone_ from
+submodules](https://git-scm.com/book/en/v1/Git-Tools-Submodules).
+This causes the common __fork and clone not to work__.
+Instead, _clone_ from
 https://github.com/SWI-Prolog/swipl-devel.git and next associate your
-clone with your _fork_ (replace `me` with your github user name).
+clone with your _fork_.
+Follow the steps below.
+
+First, clone this Git repository:
 
     git clone https://github.com/SWI-Prolog/swipl-devel.git
     cd swipl-devel
-    git submodule update --init
+    git submodule init
+
+Finally, fork this repository on GitHub, and associate your fork:
+
+    # Replace "myfork" with any name you want except "origin".
+    # Replace "me" with your github user name.
+
     git remote add myfork git@github.com:me/swipl-devel.git
 
-See [How to submit a patch](http://www.swi-prolog.org/howto/SubmitPatch.html)
-for details.
+Optional: Clone the package submodules you want to use/modify.
+(If you don't know which packages you need, skip this step.
+You can come back here later.)
+You can specify individual package paths in the command line:
 
-## Building
+    git submodule update -- packages/jpl packages/clib packages/sgml
 
-See
-[CMAKE.md](https://github.com/SWI-Prolog/swipl-devel/blob/master/CMAKE.md)
+## Building from source
+
+See [CMAKE.md](CMAKE.md)
 and [Build SWI-Prolog from source](http://www.swi-prolog.org/build/)
 
+## Contributing
+
+There are several ways to contribute to SWI-Prolog:
+
+- Write source code.
+[Submit a patch](http://www.swi-prolog.org/howto/SubmitPatch.html).
+- Write or improve documentation. (How?)
+- Report an issue. (How?)
+- Help answer questions in the [SWI-Prolog Discourse group](https://swi-prolog.discourse.group).
+- Propose improvements in the SWI-Prolog Discourse group.
+- Donate/sponsor/fund an improvement. (How?)
```

Comment:
What should write here about donation? How do we expect people to donate/sponsor/fund?

Reply:
I'm mostly happy with the changes.   I would like to keep the README as short as possible though, mostly pointing at places.   I'm in favor of changing _Contributing_ to be just a link and possibly change some stuff on the website.   The git repo is not the typical place people start as there are binaries for most platforms (I think).   I'd drop donating.   We've done that in the past.  It is mostly hassle that is nice to buy some infrastructure, but doesn't cover salaries.   So far we managed to get the required infra structure sponsored directly.

P.s.  The website is in the `plweb.git` repo with most data in `plweb-www.git`